conan/tools/qbs/generic.py

```diff
@@ -0,0 +1,296 @@
+        if compiler == 'Visual Studio':
+            if tools.msvs_toolset(conanfile) == 'ClangCl':
+                return 'clang-cl'
+            return 'cl'
+        if compiler == 'clang':
+            return 'clang-cl'
+        raise QbsException('unknown windows compiler')
+
+    return compiler
+
+
+def _settings_dir(conanfile):
+    return '%s/conan_qbs_toolchain_settings_dir' % conanfile.install_folder
+
+
+def _setup_toolchains(conanfile):
+    if tools.get_env('CC'):
+        compiler = tools.get_env('CC')
+    else:
+        compiler = _default_compiler_name(conanfile)
+
+    env_context = tools.no_op()
+    if platform.system() == 'Windows':
+        if compiler in ['cl', 'clang-cl']:
+            env_context = tools.vcvars()
+
+    with env_context:
+        cmd = 'qbs-setup-toolchains --settings-dir %s %s %s' % (
+              _settings_dir(conanfile), compiler, _profile_name)
+        conanfile.run(cmd)
+
+
+def _read_qbs_toolchain_from_config(conanfile):
+    s = StringIO()
+    conanfile.run('qbs-config --settings-dir %s --list' % (
+                    _settings_dir(conanfile)), output=s)
+    config = {}
+    s.seek(0)
+    for line in s:
+        colon = line.index(':')
+        if 0 < colon and not line.startswith('#'):
+            full_key = line[:colon]
+            if full_key.startswith(_profiles_prefix_in_config):
+                key = full_key[len(_profiles_prefix_in_config)+1:]
+                value = line[colon+1:].strip()
+                if value.startswith('"') and value.endswith('"'):
+                    temp_value = value[1:-1]
+                    if (temp_value.isnumeric() or
+                            temp_value in ['true', 'false', 'undefined']):
+                        value = temp_value
+                config[key] = value
+    return config
+
+
+class LinkerFlagsParser(object):
+    def __init__(self, ld_flags):
+        self.driver_linker_flags = []
+        self.linker_flags = []
+        self._parse = self._detect_wl_or_add_driver_linker_flag
+        for token in ld_flags:
+            self._parse(token)
+
+    def _detect_wl_or_add_driver_linker_flag(self, token):
+        if token == '-Wl':
+            self._parse = self._assert_comma
+        else:
+            self.driver_linker_flags.append(token)
+            self._parse = self._detect_wl_or_add_driver_linker_flag
+
+    def _detect_wl_or_detect_comma_or_add_driver_linker_flag(self, token):
+        if token == ',':
+            self._parse = self._add_linker_flag
+        else:
+            self._detect_wl_or_add_driver_linker_flag(token)
+
+    def _assert_comma(self, token):
+        if token != ',':
+            raise QbsException('Could not parse LDFLAGS')
+        self._parse = self._add_linker_flag
+
+    def _add_linker_flag(self, token):
+        self.linker_flags.append(token)
+        self._parse = self._detect_wl_or_detect_comma_or_add_driver_linker_flag
+
+
+def _flags_from_env():
+    flags_from_env = {}
```

Comment:
The way we are going to use environment variables in toolchains is still not defined, so this might be broken in the future. I am ok with adding it now this way, as long as it is clear that this integration is very experimental and subject to future breaking change.

Reply:
I the environment should not be used, how do you plan to pass the compiler path or cross compiling flags? The current cross compiling goto way seems like environment variables: https://docs.conan.io/en/latest/systems_cross_building/cross_building.html#conan-v1-24-and-newer